python/sglang/srt/weight_utils.py

```diff
@@ -0,0 +1,397 @@
+"""Utilities for downloading and initializing model weights."""
+import fnmatch
```

Comment:
Where is this file from? Are there any references? If so, could you put some links to the source of code here?

Reply:
@ZhouXingg  Please add comment that this is code is copied/adapted from vllm before their refractor causing sglang compat issues.